src/patterns.md

```diff
@@ -756,6 +761,57 @@ Path patterns are irrefutable when they refer to structs or an enum variant when
 has only one variant or a constant whose type is irrefutable. They are refutable when they
 refer to refutable constants or enum variants for enums with multiple variants.
 
+## Or-patterns
+
+_Or-patterns_ are patterns that match on either of two sub-patterns (e.g. `A |
+B`). They can nest arbitrarily. Syntactically, or-patterns are allowed in any
+of the places where other patterns are allowed (represented by the _Pattern_
+production), with the exceptions of function and closure arguments (represented
+by the _PatternNoTopAlt_ production). Additionally, the macro `:pat` matchers
+only match or-patterns in the 2021+ editions.
```

Comment:
I would just leave out this point about macros, and instead just document in macros-by-example.md.  Also, the reference has a style of using blockquotes to call out edition differences with `> **Edition Differences**:`, so I would expect something similar in the macros page.

```suggestion
by the _PatternNoTopAlt_ production).
```